homework-3/Program.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.IO;
+
+public class Logger : IDisposable
+{
+    private static Logger? instance; 
+    private readonly string logFilePath;
+    private StreamWriter? logWriter;
+
+    private Logger(string logFilePath)
+    {
+        this.logFilePath = logFilePath;
+        logWriter = new StreamWriter(logFilePath, true);
+    }
+
+    public static Logger Instance
+    {
+        get
+        {
+            instance ??= new Logger("log.txt");
+            return instance;
+        }
+    }
+
+    private void Log(string message)
+    {
+        logWriter?.WriteLine($"{DateTime.Now}: {message}");
+        logWriter?.Flush();
+    }
+
+    public void Dispose()

```

Comment:
this function is called no where.. this should be called in the destructor of logger class